docs/reference/troubleshoot.md

```diff
@@ -195,6 +197,70 @@ For example, for a `CloudSQLInstance` managed resource (`database.gcp.crossplane
 kubectl patch cloudsqlinstance my-db -p '{"metadata":{"finalizers": []}}' --type=merge
 ```
 
+## Handling Crossplane Package Dependency
+
+When use `crossplane.yaml` to define Crossplane package, you can specify packages that 
+the current package depends on including the version constraint using `spec.dependsOn`.
+
+Please make sure you provide the fully qualified address to the dependent package, but do
+not append the package version (i.e. the OCI image tag) after the package name. This may
+lead to the missing dependency error when Crossplane tries to install the dependency.
+
+When specify the version constraint, you should strictly follow the [semver spec](https://github.com/Masterminds/semver#basic-comparisons). Otherwise,
+it may not be able to find the appropriate version for the dependent package even it says
+the dependency is found. This may lead to the incompatible dependency error during the installation.
+
+Below is an example where a configuration package depends on a provider pulled from `crossplane/provider-aws`.
```

Comment:
can you make sure to capitalize Configuration and Package wherever talking about the package itself, and use `Configuration` and `Package` when talking about the actual k8s resources (which I believe you already do)
```suggestion
Below is an example where a Configuration package depends on a provider pulled from `crossplane/provider-aws`.
```